src/shared/lib/hooks/redux.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+
+export const useAppDispatch = () => useDispatch<ThunkAppDispatch>;
```

Comment:
We should use `AppDispatch` type here. If you intend to add a thunks middleware (or whatever middleware) you can do that and if do that the right way, types from middlewares will be correctly inferred to dispatch function.

https://redux-toolkit.js.org/usage/usage-with-typescript#correct-typings-for-the-dispatch-type